### PR TITLE
fix(linting): add scripts, docs, smoke-tests

### PR DESCRIPTION
**Pull request by wraithgar**
There is code in all of these places we control so it should be linted

### CODE REVIEW

File: scripts/bundle-and-gitignore-deps.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const arb = new Arborist({ path: resolve(__dirname, '..') })
 const shouldIgnore = []
 
 arb.loadVirtual().then(tree => {
-  for (const [name, node] of tree.children.entries()) {
+  for (const [, node] of tree.children.entries()) {
```
Comment (isaacs) [inline scripts/bundle-and-gitignore-deps.js]: consider: `for (const node of tree.children.values())`

**Review by isaacs — APPROVED** [review]
One tiny nit that I don't actually care about just to prove I actually read the thing ;)

LGTM!

**Comment by wraithgar** [comment]
> One tiny nit that I don't actually care about just to prove I actually read the thing ;)

You read it better than me, I treated the symptom not the problem.

File: scripts/bundle-and-gitignore-deps.js (hunk 1 above)
Comment (wraithgar) [reply scripts/bundle-and-gitignore-deps.js]:
```suggestion
  for (const node of tree.children.values()) {
```

**Pull request merged**